node/src/components/storage.rs

```diff
@@ -735,13 +824,13 @@ impl Storage {
                         return Ok(responder.respond(None).ignore());
                     };
                 // Check that the hash of the block retrieved is correct.
-                assert_eq!(&block_hash, block.hash());
+                debug_assert_eq!(&block_hash, block.hash());
```

Comment:
I think we should log an error here, and at the new `debug_assert!` just below too.

Reply:
Fixed in d15a3e057